tests/test_create_inconsistent_dimensionality.py

```diff
@@ -29,9 +30,29 @@
 
 
 wkt_cases = [
-    ('LINESTRING (1 1 1, 2 2)', 'LINESTRING Z (1 1 1, 2 2 0)'),
-    ('POLYGON ((0 0 0, 1 0 0, 1 1, 0 1 0, 0 0 0))', 'POLYGON Z ((0 0 0, 1 0 0, 1 1 0, 0 1 0, 0 0 0))')
+    # preserve 3rd dimension
+    ('MULTIPOINT (1 1 1, 2 2)', 'MULTIPOINT Z (1 1 1, 2 2 0)'),
+    ('MULTIPOINT (1 1, 2 2 2)', 'MULTIPOINT Z (1 1 0, 2 2 2)'),
+    ('POLYGON ((0 0 0, 1 0 0, 1 1, 0 1 0, 0 0 0))',
+     'POLYGON Z ((0 0 0, 1 0 0, 1 1 0, 0 1 0, 0 0 0))'),
+    # drop 3rd dimension
+    ('POLYGON ((0 0, 1 0 1, 1 1, 0 1, 0 0))',
+     'POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))'),
 ]
+if geos_version < (3, 9, 0):
+    wkt_cases += [
+        # preserve 3rd dimension
+        ('LINESTRING (1 1 1, 2 2)', 'LINESTRING Z (1 1 1, 2 2 0)'),
+        # drop 3rd dimension
+        ('LINESTRING (1 1, 2 2 2)', 'LINESTRING (1 1, 2 2)'),
+    ]
+else:
+    wkt_cases += [
+        # drop 3rd dimension
+        ('LINESTRING (1 1 1, 2 2)', 'LINESTRING (1 1, 2 2)'),
+        # preserve 3rd dimension
+        ('LINESTRING (1 1, 2 2 2)', 'LINESTRING Z (1 1 0, 2 2 2)'),
+    ]
```

Comment:
It seems to be quite inconsistent when an incomplete third dimension is preserved and when it is dropped, right? And also the change with GEOS 3.9 does not seem to make it more consistent

Reply:
I call this a minor annoyance; see [GEOS trac#1079](https://trac.osgeo.org/geos/ticket/1079)

Reply:
Likely fixed here https://github.com/pramsey/geos/commit/9211fae903a22fc8fc68de47a1a0357fe1bfa4b8